app/validators/measure_type_validator.rb

```diff
@@ -8,7 +8,34 @@ class MeasureTypeValidator < TradeTariffBackend::Validator
     validates :validity_dates
   end
 
+  validation :MT3,
+             'When a measure type is used in a measure then the validity period of the measure type must span the validity period of the measure.',
+             on: [:create, :update] do |record|
+    if record.measures.any?
+      record.measures.all? {|measure|
+        record.validity_start_date <= measure.validity_start_date &&
+        ((record.validity_end_date.blank? || measure.validity_end_date.blank?) ||
+          (record.validity_end_date >= measure.validity_end_date))
+      }
+    end
+  end
+
   validation :MT4, 'The referenced measure type series must exist.', on: [:create, :update] do
     validates :presence, of: :measure_type_series
   end
+
+  validation :MT7,
+             'A measure type can not be deleted if it is used in a measure.',
+             on: [:destroy] do |record|
+    record.measures.none?
+  end
+
+  validation :MT10,
+             'The validity period of the measure type series must span the validity period of the measure type.' do |record|
+    if record.measure_type_series.present?
```

Comment:
It must be present

Reply:
So we don't need this extra check`record.measure_type_series.present?` as it is always present for the measure_type record, right?

Reply:
Well, can we make sure conformance rules run in order, e.g. MT4 before MT10? then you do not the additional guard

Reply:
Removed that extra check. As we already handle it in MT4 and MT validation code is written before ME10 validation code.